analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.FlowCaptureOperation.cs

```diff
@@ -0,0 +1,354 @@
+        context.ValidateEntry(LiveIn("s1", "s2", "s3", "s4"), LiveOut("s1", "s2", "s3", "s4"));
+        context.Validate(context.Cfg.Blocks[1], null, LiveIn("s1", "s2", "s3", "s4"), LiveOut("s1", "s2", "s3", "s4"));  // 1: Parents: Entry, Children: 2
+        context.Validate(context.Cfg.Blocks[2], null, LiveIn("s1", "s2", "s3", "s4"), LiveOut("s1", "s2", "s3", "s4"));  // 2: Parents: 1, Children: 2
+        context.Validate(context.Cfg.Blocks[3], null, LiveIn("s1"), LiveOut("s1"));                                      // 3: Parents: 2, Children: 13
+        context.Validate(context.Cfg.Blocks[4], null, LiveIn("s1", "s2", "s3", "s4"), LiveOut("s1", "s2", "s3", "s4"));  // 4: Parents: 2, Children: 5
+        context.Validate(context.Cfg.Blocks[5], null, LiveIn("s1", "s2", "s3", "s4"), LiveOut("s1", "s2", "s3", "s4"));  // 5: Parents: 4, Children: 6
+        context.Validate(context.Cfg.Blocks[6], null, LiveIn("s1", "s2", "s3", "s4"), LiveOut("s1", "s2", "s3", "s4"));  // 6: Parents: 5, Children: 7, 8
+        context.Validate(context.Cfg.Blocks[7], null, LiveIn("s1", "s2", "s3"), LiveOut("s1", "s2", "s3"));              // 7: Parents: 6, Children: 12
+        context.Validate(context.Cfg.Blocks[9], null, LiveIn("s1", "s2", "s3", "s4"), LiveOut("s1", "s2", "s3", "s4"));  // 8: Parents: 6, Children: 9, 10
+        context.Validate(context.Cfg.Blocks[9], null, LiveIn("s1", "s2", "s3", "s4"), LiveOut("s1", "s2", "s3", "s4"));  // 9: Parents: 8, Children: 11
+        context.Validate(context.Cfg.Blocks[10], null, LiveIn("s1", "s2", "s3", "s4"), LiveOut("s1", "s2", "s3", "s4")); // 10: Parents: 8, Children: 11
+        context.Validate(context.Cfg.Blocks[11], null, LiveIn("s1", "s2", "s3", "s4"), LiveOut("s1", "s2", "s3"));       // 11: Parents: 9, 10, Children: 12
+        context.Validate(context.Cfg.Blocks[12], null, LiveIn("s1", "s2", "s3"), LiveOut("s1"));                         // 12: Parents: 7, 11, Children:  13
+        context.Validate(context.Cfg.Blocks[13], null, LiveIn("s1"));                                                    // 13: Parents: 3, 12, Children: Exit
+        context.ValidateExit();
+    }
+
+    [TestMethod]
+    public void FlowCaptrure_NullCoalescingOperator_Nested()
+    {
+        /*       Block 1
+         *       #0=s1
+         *     #0 is null
+         *        /   \
+         *      F/     \T
+         *      /       \
+         *   Block 2  Block 3---+------F----> Block 4 --------+
+         *   #1=#0   s2 is null |             #2=s3           |
+         *       \     /        |                             |
+         *        Block 6       +------T----> Block 5-------->|
+         *         s=#1                   #2="NestedFalse"    |
+         *          |                                         |
+         *          |                                Block6<--+
+         *          |                              #2 is null
```

Comment:
This shape is wrong. Block 3 doesn't have 3 successors and `Block6` is present twice.
Is it worth having it?

Reply:
I'll remove it and keep notes next to the assertions.
It's too monstrous.